deps/graph-parser/src/logseq/graph_parser/util/db.cljs

```diff
@@ -22,56 +22,93 @@ it will return 1622433600000, which is equivalent to Mon May 31 2021 00 :00:00."
 
+(defn old->new-relative-date-format [input]
+  (let [count (re-find #"^\d+" (name input))
+        plus-minus (if (re-find #"after" (name input)) "+" "-")
+        ms? (string/ends-with? (name input) "-ms")]
+    (keyword :today (str plus-minus count "d" (if ms? "-ms" "")))))
+
+(comment
+  (old->new-relative-date-format "1d")
+  (old->new-relative-date-format "1d-before")
+  (old->new-relative-date-format "1d-after")
+  (old->new-relative-date-format "1d-before-ms")
+  (old->new-relative-date-format "1d-after-ms")
+  (old->new-relative-date-format "1w-after-ms"))
+
+(defn keyword-input-dispatch [input]
+  (cond 
+    (#{:current-page :current-block :parent-block :today :yesterday :tomorrow :right-now-ms} input) input
+    (re-find #"^[+-]\d+[dwmy](-ms)?$" (name input)) :relative-date
+    (re-find #"^\d+d(-before|-after|-before-ms|-after-ms)?$" (name input)) :DEPRECATED-relative-date
+    (= :start-of-today-ms input) :DEPRECATED-start-of-today-ms 
+    (= :end-of-today-ms input) :DEPRECATED-end-of-today-ms))
+(defmulti resolve-keyword-input (fn [_db input _opts] (keyword-input-dispatch input))) 
+
+(defmethod resolve-keyword-input :DEPRECATED-relative-date [db input opts]
+  ;; This handles all of the cases covered by the following:
+  ;; :Xd, :Xd-before, :Xd-before-ms, :Xd-after, :Xd-after-ms
+  (resolve-keyword-input db (old->new-relative-date-format input) opts))
+
+(defmethod resolve-keyword-input :DEPRECATED-start-of-today-ms [db _ opts]
+  (resolve-keyword-input db :today/-0d-ms opts))
+
+(defmethod resolve-keyword-input :DEPRECATED-end-of-today-ms [db _ opts] 
+  (resolve-keyword-input db :today/+0d-ms opts))
+
+(defmethod resolve-keyword-input :current-page [_ _ {:keys [current-page-fn]}]
+  ;; TODO: handle current-page-fn not being provided
+  (some-> (current-page-fn) string/lower-case))
+
+(defmethod resolve-keyword-input :current-block [db _ {:keys [current-block-uuid]}]
+  ;; TODO: handle current-block-uuid not being provided
+  (:db/id (d/entity db [:block/uuid current-block-uuid])))
+
+(defmethod resolve-keyword-input :parent-block [db _ {:keys [current-block-uuid]}]
+  ;; TODO: handle current-block-uuid not being provided
+  (:db/id (:block/parent (d/entity db [:block/uuid current-block-uuid]))))
+
+(defmethod resolve-keyword-input :today [_ _ _]
+  (date->int (t/today)))
+
+(defmethod resolve-keyword-input :yesterday [_ _ _]
+  (date->int (t/minus (t/today) (t/days 1))))
+
+(defmethod resolve-keyword-input :tomorrow [_ _ _]
+  (date->int (t/plus (t/today) (t/days 1))))
+
+(defmethod resolve-keyword-input :right-now-ms [_ _ _]
+  (date-time-util/time-ms))
+
+(defmethod resolve-keyword-input :relative-date [db input opts]
+  (let [relative-to (case (or (namespace input) "today")
+                      "today" (t/today))
+        [_ offset-direction offset offset-unit offset-ms?] (re-find #"^([+-])(\d+)([dwmy])(-ms)?$" (name input))
+        offset-fn (case offset-direction "+" t/plus "-" t/minus)
```

Comment:
I'm digging the explicitness of the syntax we agreed to. This sets us up nicely in the future to support chaining these down the road :)

Reply:
What do you mean by chaining these?

Reply:
Don't want to scope creep this PR hehe, but down the road we could get asks to add or subtract multiple date ranges so `+1w+1d` or `+1y-1w` would naturally fit